dns/src/lib.rs

```diff
@@ -249,7 +249,7 @@ impl<'a, W5500: Udp> Response<'a, W5500> {
     /// Number of answers in the response.
     #[must_use]
     pub fn answer_count(&self) -> u16 {
-        self.header.ancount()
+        self.header.ancount() + self.header.arcount()
```

Comment:
I have not looked too much at the RFC in this area.  Is there a reason this is just `answer + additional`; should it be changed to return the total number of resource records, including `NSCOUNT` too?

```rs
/// Number of resource records in the response.
#[must_use]
pub fn rr_count(&self) -> u16 {
    self.header.ancount().saturating_add(self.header.nscount()).saturating_add(self.header.arcount())
}
```

`next_answer` should be changed to `next_rr` too if the above change is made.

Reply:
I've not looked at it at all, I've mostly been working off what I see in Wireshark.